src/audio/CMakeLists.txt

```diff
@@ -163,7 +163,12 @@ check_optimization(hifi3 -mhifi3 "" -DOPS_HIFI3)
 set(sof_audio_modules mixer volume src asrc eq-fir eq-iir dcblock crossover tdfb drc multiband_drc mfcc mux)
 
 # sources for each module
-set(volume_sources module_adapter/module/volume/volume.c module_adapter/module/volume/volume_generic.c)
+if(CONFIG_IPC_MAJOR_3)
+	set(volume_sources module_adapter/module/volume/volume_ipc3.c module_adapter/module/volume/volume_generic.c)
+elseif(CONFIG_IPC_MAJOR_4)
+	set(volume_sources module_adapter/module/volume/volume_ipc4.c module_adapter/module/volume/volume_generic.c)
+endif()
+
```

Comment:
Looks like sources are the same except the ipc file. Does cmake support creating a ```volume_sources``` with common files and then only *add* ipc3/4 depending on the Kconfig ?

Reply:
source will not be same in following patches, this patch just create ipc3 and ipc4 source file, following patches will remove each unused code, and common code out, please stay tuned.